roles/cleanup_gs_data/templates/cleanup_gs.bash

```diff
@@ -1,73 +1,92 @@
-# If folder is empty for 2 weeks -> delete folder
-# If folder is not empty, but only a  .finished file is present (2 weeks) -> delete folder
-# If folder is not empty and no .finished file is present (1 week) -> notification, 2 weeks -> delete folder
-
 #!/bin/bash
 
-dirToCheck="/groups/umcg-genomescan/"*
+slurm_notification_slack_webhook='{{ slurm_notification_slack_webhook }}'
+{% raw %}
+dirToCheck="/groups/umcg-genomescan/"
 dateInSecNow=$(date +%s)
 
-# Check only dirs, ignore files and reverse sort to check subfolders first
-for dir in $(find ${dirToCheck} -maxdepth 1 -type d | sort -r)
+#
+# Find all GenomeScan batch dirs ignoring files.
+#
+# If folder is older than 2 weeks -> delete folder.
+# If folder is younger
+#    If it only contains leftovers from already processed data -> do nothing.
+#    If it contains unprocessed data and is older than 1 week -> send warning.
+#
+# Leftovers from processed data sets:
+#   Old style with only raw data:
+#     [0-9]{6}-[0-9]{3}/
+#     [0-9]{6}-[0-9]{3}/[0-9]{6}-[0-9]{3}.finished
+#   E.g.:
+#     104832-210/
+#     104832-210/104832-210.finished
+#   New style Dragen data:
+#     [0-9]{6}-[0-9]{3}/
+#     [0-9]{6}-[0-9]{3}/[0-9]{6}-[0-9]{3}.finished
+#     [0-9]{6}-[0-9]{3}/Analysis/
+#     [0-9]{6}-[0-9]{3}/Raw_data/
+#     [0-9]{6}-[0-9]{3}/UMCG_CSV_[0-9]{6}-[0-9]{3}.csv
+#   E.g.
+#     104832-210/
+#     104832-210/104832-210.finished
+#     104832-210/Analysis/
+#     104832-210/Raw_data/
+#     104832-210/UMCG_CSV_104832-210.csv
+#
+for dir in $(find "${dirToCheck}" -mindepth 1 -maxdepth 1 -type d | sort)
 do
-	#Check ctime instead of mtime and remove /groups from dir to be able to look it up is fs
+	echo "Processing ${dir} ..."
+	#
+	# Check creation time instead of modification time
+	# and remove /groups from dir for lookup with debugfs.
+	#
 	creationTime=$(/sbin/debugfs -R 'stat '${dir#*/*/} /dev/vdb | awk '/crtime/{print $2}' FS='--')
```

Comment:
I know this is old line, but still:
are we sure we always have this data on ext2/3/4 and never on xfs?
Because `debugfs` man states that it is supporting ext filesystems.

Reply:
Indeed `debugfs` only works for `ext` file systems and parsing the _crtime_ a.k.a. _birth time_ is a bit of a brittle construct. Now that we have all _data transfer_ servers on Rocky 9 the kernels support reporting this value using the `stat` command. I will update the code. (Previously on CentOS 7.x the kernel was too old and `stat` could not report `crtime`.)